Sources/LLVM/ORCJIT.swift

```diff
@@ -0,0 +1,247 @@
+    var target: LLVMOrcTargetAddress = 0
+    let err = LLVMOrcGetSymbolAddress(jit.llvm, &target, symbol)
+    guard err == LLVMOrcErrSuccess else {
+      return nil
+    }
+    return target
+  }
+}
+
+/// Runs a series of symbol resolvers in order, stopping when the first resolver
+/// finds a valid symbol.
+public struct CompoundSymbolResolver: SymbolResolver {
+  private let resolverChain: [SymbolResolver]
+
+  /// Create a symbol resolver that will run the provided resolvers in order
+  /// until a symbol is resolved.
+  ///
+  /// - Parameter resolvers: The resolvers to run. Each one will be queried for
+  ///                        each address they're resolving -- if they return
+  ///                        a non-nil value, then the resolution stops and
+  ///                        that result is returned. If none of the provided
+  ///                        resolvers return an address, then this resolver
+  ///                        returns no address.
+  public init(resolvers: [SymbolResolver]) {
+    self.resolverChain = resolvers
+  }
+
+  /// Create a symbol resolver that will run the provided resolvers in order
+  /// until a symbol is resolved.
+  ///
+  /// - Parameter resolvers: The resolvers to run. Each one will be queried for
+  ///                        each address they're resolving -- if they return
+  ///                        a non-nil value, then the resolution stops and
+  ///                        that result is returned. If none of the provided
+  ///                        resolvers return an address, then this resolver
+  ///                        returns no address.
+  public init(resolvers: SymbolResolver...) {
+    self.init(resolvers: resolvers)
+  }
+
+  public func address(of symbol: String, in jit: ORCJIT) -> UInt64? {
+    for resolver in resolverChain {
+      if let addr = resolver.address(of: symbol, in: jit) { return addr }
+    }
+    return nil
+  }
+}
+
+
+/// A C-compatible function handler that wraps a JIT's symbol resolution
+/// callback.
+///
+/// - Parameters:
+///   - symbol: The symbol being looked up.
+///   - ctx: A void pointer pointing to a retained ORCJIT instance.
+/// - Returns: A callable address of a compiled function, or 0 if no such
+///            function was found.
+func orcjitSymbolResolver(symbol: UnsafePointer<Int8>?,
+                          ctx: UnsafeMutableRawPointer?) -> UInt64 {
+  guard let ctx = ctx else { return 0 }
+  let handle = Unmanaged<ORCJIT>.fromOpaque(ctx)
+  guard let cString = symbol else { return 0 }
+  let jit = handle.takeRetainedValue()
+  return jit.resolver.address(of: String(cString: cString), in: jit) ?? 0
+}
+
+public class ORCJIT {
+  let llvm: LLVMOrcJITStackRef
+  let resolver: SymbolResolver
+
+  /// The strategy that ORCJIT should use to compile a specific module.
+  public enum CompilationStrategy {
+    /// The IR will be compiled lazily, one function at a time.
+    case lazy
```

Comment:
The JIT provides `LLVMOrcCreateLazyCompileCallback` to be fired when it resolves symbols.  We could wrap this and let the user pass us a block.